src/crewai/tools/tool_usage.py

```diff
@@ -203,6 +206,21 @@ def _use(
             attempts=self._run_attempts,
         )
         result = self._format_result(result=result)  # type: ignore # "_format_result" of "ToolUsage" does not return a value (it only ever returns None)
+        data = {
+            "result": result,
+            "tool_name": tool.name,
+            "tool_args": calling.arguments,
+        }
+
+        if (
+            hasattr(original_tool, "result_as_answer")
```

Comment:
Ideally, we should setup a interface/protocol in crewAI-tools, so tools would optionally respect the protocol and we could check for it instead of searching for an attribute. Happy to take care of this as a follow up PR.

Reply:
<3 Love it good idea